core/state/pruner/pruner.go

```diff
@@ -233,6 +256,96 @@ func prune(snaptree *snapshot.Tree, root common.Hash, maindb ethdb.Database, sta
 	return nil
 }
 
+func (p *BlockPruner) backUpOldDb(name string, cache, handles int, freezer, namespace string, args ...bool) ([]*types.Block, []types.Receipts, []*big.Int, error) {
+	chainDb, err := p.n.OpenDatabaseWithFreezer(name, cache, handles, freezer, namespace, args...)
+	if err != nil {
+		log.Error("Failed to open ancient database: %v", err)
+		return nil, nil, nil, err
+	}
+	defer chainDb.Close()
+
+	// Get the number of items in ancient db, frozen it is.
+	frozen, err := chainDb.Ancients()
+	// Write the latest 128 blocks data of the ancient db
+	// If we can't access the freezer or it's empty, abort.
+	if err != nil || frozen == 0 {
+		log.Error("can't access the freezer or it's empty, abort")
+		return nil, nil, nil, errors.New("can't access the freezer or it's empty, abort")
+	}
+
+	oldOffSet := rawdb.ReadOffSetOfAncientFreezer(chainDb)
+	// Get the actual start block number.
+	startBlockNumber := frozen - blockPruneBackUpBlockNumber + oldOffSet
+
+	// Initialize the slice to buffer the 128 block data.
+	blockList := make([]*types.Block, 0, blockPruneBackUpBlockNumber)
+	receiptsList := make([]types.Receipts, 0, blockPruneBackUpBlockNumber)
+	externTdList := make([]*big.Int, 0, blockPruneBackUpBlockNumber)
+	// All ancient data within the most recent 128 blocks write into memory buffer for future new ancient_back directory usage.
+	for blockNumber := startBlockNumber; blockNumber < frozen+oldOffSet; blockNumber++ {
+		blockHash := rawdb.ReadCanonicalHash(chainDb, blockNumber)
+		block := rawdb.ReadBlock(chainDb, blockHash, blockNumber)
+		blockList = append(blockList, block)
+		receipts := rawdb.ReadRawReceipts(chainDb, blockHash, blockNumber)
+		receiptsList = append(receiptsList, receipts)
+		// Calculate the total difficulty of the block
+		td := rawdb.ReadTd(chainDb, blockHash, blockNumber)
+		if td == nil {
+			return nil, nil, nil, consensus.ErrUnknownAncestor
+		}
+		externTd := new(big.Int).Add(block.Difficulty(), td)
+		externTdList = append(externTdList, externTd)
+	}
+
+	// For every round, newoffset actually equals to the startBlockNumber in ancient db.
+	newOffSet := oldOffSet + frozen - 128
+	// Write the new offset into db for the future new freezer usage.
+	rawdb.WriteOffSetOfAncientFreezer(chainDb, newOffSet)
+	return blockList, receiptsList, externTdList, nil
+}
+
+// Backup the ancient data for the old ancient db, i.e. the most recent 128 blocks in ancient db.
+func (p *BlockPruner) BlockPruneBackUp(name string, cache, handles int, namespace string, readonly bool) error {
+
+	start := time.Now()
+	isBlockPruner := true
+
+	blockList, receiptsList, externTdList, err := p.backUpOldDb(name, cache, handles, p.oldAncientPath, namespace, readonly, isBlockPruner)
```

Comment:
`isBlockPruner` is always true and `backUpOldDb` is called only here, why bother add `args`

Reply:
Done